src/explore-education-statistics-common/src/modules/table-tool/components/FiltersForm.tsx

```diff
@@ -162,12 +161,32 @@ export default function FiltersForm({
       return [];
     });
 
+    const initialFilterHierarchyValues = Object.fromEntries(
+      // Set up intitial value for each hierachy,
+      // include initialValue filters if they're in the hierarchy
+      filterHierarchies.map(tiers => {
+        const hierachyKey = tiers.at(-1)!.childFilterId;
+
+        const hierarchyOptionValues = tiers
+          .flatMap(tier => Object.values(tier.hierarchy))
+          .flat()
+          .concat(Object.keys(tiers[0].hierarchy));
+
+        return [
+          hierachyKey,
+          initialValues?.filters.filter(
+            optionId => !hierarchyOptionValues.includes(optionId),
+          ) ?? [],
+        ];
+      }),
+    );
+
```

Comment:
This is the fix for the filter hierarchies, when you'd go back to the previous step and it would select Total for you. A lack of initial value was causing the issue